chore: dynamic plugin imports

ChristopherTrimboli opened the pull request:
# Relates to:

None, just looking for improvements. 

# Risks

HIGH - could break plugins

# Background

Plugins even if unused are being always imported into the agent/index.ts at top of file.
This adds much memory usage to the JS runtime and not most efficient way to load optional plugins.

## What does this PR do?

Dynamically imports plugins when needed based on ENV / secrets loaded.

## What kind of change is this?

Improvements (misc. changes to existing features)

## Why are we doing this? Any context or related work?

Makes Eliza faster / slimmer / less bloated / scales infinite plugins with minimal runtime overhead.

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

Try out different combinations of plugins off and on, see if loads or breaks.

## Discord username

cjft

ChristopherTrimboli commented:
P.S. - seems like passed all tests, omg, nice... just noticed could do same for all client packages: if we like this change, ill do another PR for dynamic client imports too, massive optz.

ChristopherTrimboli commented:
hmmm avaer told me that "alot of wrappings" I can make better abstraction for loading dynamic imports... I'll work on. Like a function for it instead.

ChristopherTrimboli commented:
> hmmm avaer told me that "alot of wrappings" I can make better abstraction for loading dynamic imports... I'll work on. Like a function for it instead. Bit complex because of all the diff conditionals / import styles / 1 is a default import. I'll see if can do nicely.

done, now have `loadPlugin()` abstraction with very simple string based plugin config.

ryanleecode commented:
Why do it like this? Why not pass a callback to loadPlugin that is `async () => import('my plugin')

ryanleecode commented:
The probably is the public api is purely string base its not type safe where as if you do callback import and you don't have the dependency installed, typescript will not compile it.

odilitime commented:
Probably need to update https://github.com/elizaOS/eliza/blob/4c658d7d70433fdcb2feeffe879429eaef10685d/scripts/dev.sh#L45

for future plugin devs

ChristopherTrimboli commented:
> Why do it like this? Why not pass a callback to loadPlugin that is `async () => import('my plugin')`

hm, I see point, I guess my initial version: 

https://github.com/elizaOS/eliza/pull/1383/commits/9d967c827b65c1a6a6c3a566cca93b06b529a9f8

we prefer? or some inbetween?

ChristopherTrimboli commented:
OK yeah callback param is better, I'll update.

ChristopherTrimboli commented:
changed to callback params: https://github.com/elizaOS/eliza/pull/1383/commits/2efee641eb81c7570d78d045bb1ce6b7334737c0

ChristopherTrimboli commented:
> Probably need to update
> 
> https://github.com/elizaOS/eliza/blob/4c658d7d70433fdcb2feeffe879429eaef10685d/scripts/dev.sh#L45
> 
> for future plugin devs

https://github.com/elizaOS/eliza/pull/1383/commits/c5d4dc494675f11d370500d087273c7294a2ecee updated plugin instructions

ryanleecode commented:
This doesn't work your `loadPlugin` returns null but the function is typed as `Promise<any>` where as the constructor parameter only takes `Plugin[]`. Giving it a null plugin will definately fail. 

IMO the correct approach would be something as follows.

1. Create a dynamic Plugin type such as

```ts
export type DynamicPlugin = {
    readonly factory: () => Promise<Plugin>;
    readonly shouldInit: (character: Character) => boolean;
};
```

2. Make the AgenRuntime constructor private and create a static factory function that is async and accepts `ReadonlyArray<Plugin | DynamicPlugin>` as a parameter. Await the dynamic parameters in the async static factory. Bonus points for adding `_tag: 'Plugin'` and `_tag: 'DynamicPlugin` to the respective types, although this would be a high impact change since it would fuck over downstream plugins.

ChristopherTrimboli commented:
> This doesn't work your `loadPlugin` returns null but the function is typed as `Promise<any>` where as the constructor parameter only takes `Plugin[]`. Giving it a null plugin will definately fail.
> 
> IMO the correct approach would be something as follows.
> 
> 1. Create a dynamic Plugin type such as
> 
> ```ts
> export type DynamicPlugin = {
>     readonly factory: () => Promise<Plugin>;
>     readonly shouldInit: (character: Character) => boolean;
> };
> ```
> 
> 2. Make the AgenRuntime constructor private and create a static factory function that is async and accepts `ReadonlyArray<Plugin | DynamicPlugin>` as a parameter. Await the dynamic parameters in the async static factory. Bonus points for adding `_tag: 'Plugin'` and `_tag: 'DynamicPlugin` to the respective types, although this would be a high impact change since it would fuck over downstream plugins.

I see, more complex with types then I originally thought, will think about / work on, thanks for feedback.

ryanleecode commented:
> > This doesn't work your `loadPlugin` returns null but the function is typed as `Promise<any>` where as the constructor parameter only takes `Plugin[]`. Giving it a null plugin will definately fail.
> > IMO the correct approach would be something as follows.
> > 
> > 1. Create a dynamic Plugin type such as
> > 
> > ```ts
> > export type DynamicPlugin = {
> >     readonly factory: () => Promise<Plugin>;
> >     readonly shouldInit: (character: Character) => boolean;
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. Make the AgenRuntime constructor private and create a static factory function that is async and accepts `ReadonlyArray<Plugin | DynamicPlugin>` as a parameter. Await the dynamic parameters in the async static factory. Bonus points for adding `_tag: 'Plugin'` and `_tag: 'DynamicPlugin` to the respective types, although this would be a high impact change since it would fuck over downstream plugins.
> 
> I see, more complex with types then I originally thought, will think about / work on, thanks for feedback.

Actually I stand mistaken since there is a `filter(Boolean)` and the end the nulls get filtered out. So this would work but I don't think we need the third parameter b/c u can do `import("whatever").then((mod) => mod.myPlugin)` which is also type safe.

ryanleecode commented:
Few more comments.

1. probably export the loadPlugin function
2. create another function `loadPlugins`, and then i can just spread the resulting array

ryanleecode commented:
Have you run `pnpm build` in the root of the monorepo?

ChristopherTrimboli commented:
> Have you run `pnpm build` in the root of the monorepo?

ah, fixed, OK everything compiling / working great. Gonna work on:

1.) export the loadPlugin function
2.) create another function loadPlugins, and then i can just spread the resulting array

to cleanup and ready for review / testing. thanks for help! TS god

ryanleecode commented on agent/src/index.ts:
```diff
@@ -495,6 +473,18 @@ export async function createAgent(
         );
     }
```
We can nuke this goatPlugin i don't think its used

ChristopherTrimboli replied:
nuked: https://github.com/elizaOS/eliza/pull/1383/commits/12d8537c179f2201d1d3eb5da9afd39cdd3f3ba9

ChristopherTrimboli commented:
I went big boy AAA on this one: 

https://github.com/elizaOS/eliza/pull/1383/commits/23cb0806137e8abc08913bddbb679d301377ea9e

I think this proper now. ready for review kind citizen.

ryanleecode commented on packages/core/src/plugins.ts:
```diff
@@ -0,0 +1,50 @@
+import { Character, DynamicPlugin, Plugin } from "./types";
+
+/**
+ * Gets a secret from the character settings or environment variables
+ * @param {Character} character - The character object
+ * @param {string} secret - The secret key
+ * @returns {string|undefined} The secret value or undefined if not found
+ */
```
nit: personally would remove type information in the JSDocs since its completely redudant when we are using typescript

ryanleecode commented on packages/core/src/plugins.ts:
```diff
@@ -0,0 +1,50 @@
+    character: Character,
+    secret: string
+): string | undefined {
+    return character.settings?.secrets?.[secret] || process.env[secret];
```
```suggestion
    return character.settings?.secrets?.[secret] ?? process.env[secret];
```

ChristopherTrimboli commented:
missing some plugins from all the refactors... reviewing... sec

ryanleecode commented on packages/core/src/plugins.ts:
```diff
@@ -0,0 +1,50 @@
+    character: Character,
+    definitions: DynamicPlugin[]
+): Promise<Plugin[]> {
+    const plugins: Plugin[] = [];
+    for (const def of definitions) {
+        const p = await loadPlugin(def.secrets, character, def.importFn);
+        if (p) plugins.push(p);
+    }
+    return plugins;
```
would use `Promise.all` here since dynamic imports are IO operations, hence using parallel promises is preferred.

ryanleecode commented on agent/src/index.ts:
```diff
@@ -488,12 +462,133 @@ export async function createAgent(
+    const dynamicPlugins = await loadPlugins(character, [
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.coinbaseMassPaymentsPlugin
+                ),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then((m) => m.tradePlugin),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.tokenContractPlugin
+                ),
+        },
+        {
+            secrets: ["COINBASE_API_KEY", "COINBASE_PRIVATE_KEY"],
+            importFn: () =>
+                import("@elizaos/plugin-coinbase").then(
+                    (m) => m.advancedTradePlugin
+                ),
```
I was thinking a helper function (my idea of load plugins) which is just `import('@elizaos/plugin-coinbase').then((mod) => [mod.pluginA, modPluginB, modPluginC])`. Then we don't need to import coinbase 3 times.

ryanleecode replied:
And then to add them you can just spread:

```ts
plugins: [
...myCoinbasePluginsDynamicallyImportedOrNull
]
```

ChristopherTrimboli replied:
ok this final point I think, taking break, will work on that, we getting close 👍🏻

ChristopherTrimboli replied:
This not so simple with diff conditional Coinbase secrets, not all packages rely on same secrets, so can't just import [mod.pluginA, modPluginB, modPluginC] in 1 helper I think, still think need the secret definitions unique to each module, playing with `importModuleWithCache()` solution:

```
const moduleCache: { [key: string]: Promise<Plugin> } = {};

export async function importModuleWithCache(
    moduleName: string
): Promise<Plugin | null> {
    let module = null;
    if (!moduleCache[moduleName]) {
        module = await import(moduleName)
        moduleCache[moduleName] = module;
    }
    return module;
}
``` 

but loses types, tried generics but also bad cause then need to define ` interface CoinBasePlugins`. ugh.

ChristopherTrimboli replied:
solved importCache I believe: https://github.com/elizaOS/eliza/pull/1383/commits/32c7e5a1ad3d5893a806166a4921ca218acb7c0b

Main issue was handling unique secrets, I do like the simple statements of `secrets` and `importFn`, now if duplicate importFn, it uses a cache, so won't import duplicates, but DX is much nicer else would have big nested ternaries and if/else stuffed in array with secrets spread. 

If have better style lemme know.

ChristopherTrimboli replied:
In case that secrets are exact same, uses your suggestion, as array to import: https://github.com/elizaOS/eliza/pull/1383/commits/430dab17023b16a95253ddae952578406cc4a8cf

But case still stands I think best separated if secrets are unique, for webhook, and others.
https://github.com/elizaOS/eliza/pull/1383/files#diff-935219608f7b5ca6c8b8548cfdce88c7d3cdb6bb6d9f9d8df644b364f6557e4eR438-R459

ryanleecode commented on agent/src/index.ts:
```diff
@@ -488,12 +462,151 @@ export async function createAgent(
+            importFn: () =>
+                import("@elizaos/plugin-story").then((m) => m.storyPlugin),
+        },
+        ...(teeMode === TEEMode.OFF &&
```
make this just ` === "OFF"`

ryanleecode commented on agent/src/index.ts:
```diff
@@ -25,36 +25,14 @@ import {
-import { TEEMode, teePlugin } from "@elizaos/plugin-tee";
-import { tonPlugin } from "@elizaos/plugin-ton";
-import { zksyncEraPlugin } from "@elizaos/plugin-zksync-era";
+import { TEEMode } from "@elizaos/plugin-tee";
```
Importing this will import the whole plugin unfortauntely :(

0xCardinalError commented:
Nice PR.  On the topic of plugins, I am thinking about what would be good solution in long term for plugins, as there will be hundreds of them.  It makes no sense to import them all or even have them in code. I like the approach of just adding them in .json file for each character. So maybe we can have some core plugins and just contrib plugins where it can be whatever.

odilitime commented:
If I fix the conflicts is this ready to merge?

ChristopherTrimboli commented:
> Nice PR. On the topic of plugins, I am thinking about what would be good solution in long term for plugins, as there will be hundreds of them. It makes no sense to import them all or even have them in code. I like the approach of just adding them in .json file for each character. So maybe we can have some core plugins and just contrib plugins where it can be whatever.

true some theory crafting there, but issue is types need to be correct in build, I orginally made some string solution but does not validate `Plugin` type in TS properly, ryanleecode roasted me, I think .json for plugins may not workout.

0xCardinalError commented:
> > Nice PR. On the topic of plugins, I am thinking about what would be good solution in long term for plugins, as there will be hundreds of them. It makes no sense to import them all or even have them in code. I like the approach of just adding them in .json file for each character. So maybe we can have some core plugins and just contrib plugins where it can be whatever.
> 
> true some theory crafting there, but issue is types need to be correct in build, I originally made some string solution but does not validate `Plugin` type in TS properly, ryanleecode roasted me, I think .json for plugins may not workout.

I found it perfectly suitable that char is build with .json and then we just add something like

```
    "plugins": [
        "@elizaos/plugin-dexscreener", ""@elizaos/plugin-geckio","
    ],
    ```
    
    to it and list plugins we want to load besides "core" ones.

ChristopherTrimboli commented:
> > > Nice PR. On the topic of plugins, I am thinking about what would be good solution in long term for plugins, as there will be hundreds of them. It makes no sense to import them all or even have them in code. I like the approach of just adding them in .json file for each character. So maybe we can have some core plugins and just contrib plugins where it can be whatever.
> > 
> > 
> > true some theory crafting there, but issue is types need to be correct in build, I originally made some string solution but does not validate `Plugin` type in TS properly, ryanleecode roasted me, I think .json for plugins may not workout.
> 
> I found it perfectly suitable that char is build with .json and then we just add something like
> 
> ```
>     "plugins": [
>         "@elizaos/plugin-dexscreener", ""@elizaos/plugin-geckio","
>     ],
>     ```
>     
>     to it and list plugins we want to load besides "core" ones. 
> ```

that won't build tho, still need actual imports in TS, else no type checks, i would need to see more of this in action, that is just listing strings, but is not type safe / compile safe

ChristopherTrimboli commented:
> > > Nice PR. On the topic of plugins, I am thinking about what would be good solution in long term for plugins, as there will be hundreds of them. It makes no sense to import them all or even have them in code. I like the approach of just adding them in .json file for each character. So maybe we can have some core plugins and just contrib plugins where it can be whatever.
> > 
> > 
> > true some theory crafting there, but issue is types need to be correct in build, I originally made some string solution but does not validate `Plugin` type in TS properly, ryanleecode roasted me, I think .json for plugins may not workout.
> 
> I found it perfectly suitable that char is build with .json and then we just add something like
> 
> ```
>     "plugins": [
>         "@elizaos/plugin-dexscreener", ""@elizaos/plugin-geckio","
>     ],
>     ```
>     
>     to it and list plugins we want to load besides "core" ones. 
> ```

got clarification, forgot Eliza is multi-agent support, understood, will handle the plugin in .json and active for all loaded characters in dynamic import style, not too bad update, coming soon.

bertux committed:
feat: add Arthera plugin from EVM plugin model

feat: rename plugin from EVM to Arthera and update README

feat: remove bridge and swap actions from Arthera plugin

feat: update transfer examples and templates to use AA instead of ETH

feat: update viem dependency to version 2.21.58 and adjust pnpm-lock.yaml

feat: remove unused LiFi dependencies and clean up type definitions in Arthera plugin

feat: remove bridge actions and templates from Arthera plugin

feat: remove swap actions and templates from Arthera plugin

feat: update EVM naming to Arthera

feat: update README and types for Arthera mainnet integration

feat: update plugin to use Arthera instead of mainnet

fix: add required devDependencies

fix: remove switchChain

fix: update _options type to Record<string, unknown> in transferAction

fix: correct log message format in transfer action to include wallet client address

test: enhance transfer tests with additional wallet provider and address validation

Plugin arthera merge (#3)

* feat: added arthera to default character and agent

* feat: renamed EVM_PRIVATE_KEY by ARTHERA_PRIVATE_KEY

* fix: roll back core package

* fix: workspace: version

---------

Co-authored-by: Arthera Node <node@arthera.bask.guru>

fix: run transfer test only if private key provided

fix: add missing newline at end of package.json and tsconfig.json files

marcellodesales committed:
:bug: generating package-lock.json for docusaurus updates

marcellodesales committed:
:whale: Just changing Dockerfile.docs with build comments

marcellodesales committed:
:scroll: update package-lock.json with current docs package.json

marcellodesales committed:
:scroll: update pnpm-lock.yaml with docusaurus changes

rawgroundbeef committed:
fix(solana token provider): await cached data

bitcoinbender committed:
Update .env.example

Commit:
fix: add default export to plugin-image-generation

mcp97 committed:
Merge pull request #1831 from jonathanmv/develop

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1829 from bitcoinbender/patch-1

mcp97 committed:
Merge branch 'develop' into refactor/readme

affaan-m committed:
chore: update bootstrap plugin export

Relates to:
Plugin export collisions in character files

Risks
Low - This is a minor bug fix that ensures proper module exports

Background
What does this PR do?
Ensures the bootstrap plugin has a proper default export at the bottom to avoid collisions when used in the actions module in character files. This maintains consistency with other plugins in the codebase that follow the same pattern (like the EVM plugin).

What kind of change is this?
Bug fixes (non-breaking change which fixes an issue)

Documentation changes needed?
My changes do not require a change to the project documentation.

Testing
Where should a reviewer start?
Check packages/plugin-bootstrap/src/index.ts to verify the proper export structure
Verify that the plugin can be properly imported in character files without collisions
Detailed testing steps
Load a character file that uses the bootstrap plugin
Verify that no export collisions occur when the plugin is loaded
Verify that all plugin actions are properly accessible
Deploy Notes
No special deploy steps required. This is a straightforward bug fix that maintains backward compatibility.

Discord:
affaanmustafa

mcp97 committed:
Merge branch 'develop' into develop

ai16z-demirix committed:
improvement: replacing console.log with elizaLogger.

ai16z-demirix committed:
Merge branch 'develop' into improvement/eliza-logger

web3isthefuture committed:
Update README_KOR.md

Modified the previous sentence to align with the meaning in Korean.

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1815 from gmh5225/develop

mcp97 committed:
Merge branch 'develop' into improvement/eliza-logger

mcp97 committed:
Merge pull request #1834 from ai16z-demirix/improvement/eliza-logger

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge branch 'develop' into refactor/readme

ai16z-demirix committed:
Revert "refactor: (draft) improving API error handling for coinbase integration"

This reverts commit 3a9e84e538b9253d567aa32bde3bb9d7b67ba274.

Commit:
correct faq docs link in both contribution md files

mcp97 committed:
Merge branch 'develop' into patch-4

ai16z-demirix committed:
test: fixing generation.test.ts. Adding tests for memory, goals and provider

ai16z-demirix committed:
Merge branch 'develop' into test/goals-memory-provider

MacsDickinson committed:
Merge branch 'develop' into 1838--fix-faq-link-in-contributing-md

poodengDev committed:
Fix faq link on contributing.md

poodengDev committed:
Update contributing link aswell to faq

poodengDev committed:
fix broken links

shakkernerd committed:
Merge branch 'develop' into 1838-fix-faq-link-mtoor

tcm390 committed:
sort timelines based on the llm score

Gajesh2007 committed:
fix: fixed the anthropic regex

Gajesh2007 committed:
Merge branch 'main' of https://github.com/Gajesh2007/eliza

Gajesh2007 committed:
Merge main: Add reclaim adapter and resolve conflicts

Gajesh2007 committed:
fix: conflicts

Gajesh2007 committed:
Merge branch 'develop' into main

aisjdoaisjdoi2123 committed:
Update .env.example

lalalune committed:
Merge pull request #1835 from IterateFast-Labs/develop

Update README_KOR.md

lalalune committed:
Merge pull request #1832 from Valetek/refactor/readme

refactor: typo in readme

lalalune committed:
Merge pull request #1847 from manotoor/1838-fix-faq-link-mtoor

Fix faq link on contributing.md

lalalune committed:
Merge pull request #1787 from RobertSloan22/patch-1

Create README.md

lalalune committed:
Merge pull request #1827 from macfly-base/main

Update c3po.character.json

mcp97 committed:
Merge branch 'develop' into test/goals-memory-provider

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Merge branch 'develop' into develop

lalalune committed:
Merge branch 'feat/genlayer-plugin' of https://github.com/yeagerai/eliza into HEAD

daniel-farina committed:
Add Stargaze plugin

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'bugfix/broken-ci-docs-missed-frozen' of https://github.com/marcellodesales/elizaOS into HEAD

lalalune committed:
Merge branch 'patch-4' of https://github.com/affaan-m/eliza into HEAD

lalalune committed:
Merge branch '1838--fix-faq-link-in-contributing-md' of https://github.com/MacsDickinson/eliza into HEAD

lalalune committed:
Merge branch 'develop' of https://github.com/inferanetwork/eliza into HEAD

lalalune committed:
Merge branch 'main' of https://github.com/robin-rrt/eliza into HEAD

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'develop' of https://github.com/blockydevs/eliza into HEAD

lalalune committed:
Merge branch 'roshan/plugin-tee-marlin' of https://github.com/marlinprotocol/eliza into HEAD

lalalune committed:
fix lock

odilitime committed:
Add DAO donation ask & dev discord

odilitime committed:
Merge pull request #1867 from elizaOS/odilitime-patch-1

docs: Add DAO donation ask & dev discord

tcm390 committed:
clean code and add comment

tcm390 committed:
add MAX_ACTIONS_PROCESSING

tcm390 committed:
Merge branch 'main' into tcm-max-actions

tcm390 committed:
remove unused variable

tcm390 committed:
remove unused variable

tcm390 committed:
add EmbeddingModelSettings type

tcm390 committed:
get the info from model settings instead of hardcoced number

tcm390 committed:
remove unused variable

tcm390 committed:
fix provider path

tcm390 committed:
Merge branch 'main' into tcm-refactor-modelConfig

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
fix bad merge

odilitime committed:
revert 1834 changes

odilitime committed:
audit fix

shakkernerd committed:
Merge pull request #1872 from elizaOS/fix-postinstall-script

fix: Fix postinstall script

shakkernerd committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
Merge pull request #1770 from pedronovikovborges/docs/readme-ro

docs: add Romanian README translation

tcm390 committed:
resolve conflict

shakkernerd committed:
Merge branch 'develop' into test/goals-memory-provider

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

shakkernerd committed:
Merge pull request #1840 from ai16z-demirix/test/goals-memory-provider

test: adding more tests for goals,memory and provider. Fixing generation.test.ts

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

odilitime committed:
Merge pull request #1782 from augchan42/fix/twitter-profile-cache-1638

fix: remove twitter profile caching (#1638)

shakkernerd committed:
Merge branch 'develop' into add-devcontainer

shakkernerd committed:
Merge pull request #1807 from jazzvaz/add-devcontainer

feat: Add devcontainer

shakkernerd committed:
Merge branch 'develop' into fix/await-cached-prices-2

shakkernerd committed:
Merge pull request #1828 from bentatum/fix/await-cached-prices-2

fix(solana token provider): await cached data

odilitime committed:
support more debians distros

odilitime committed:
Merge branch 'develop' into fix-postinstall-script

shakkernerd committed:
Merge pull request #1875 from odilitime/fix-postinstall-script

chore: support more debians distros

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

shakkernerd committed:
Merge pull request #1805 from elizaOS/tcm-refactor-modelConfig

feat: refactor model config

rawgroundbeef committed:
chore: enable strict null checks

ai16z-demirix committed:
test: adding test setup for telegram client

ai16z-demirix committed:
test: adding test setup for telegram client - improving message manager tests

ai16z-demirix committed:
Merge branch 'develop' into test/client-telegram

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 8" notes

kylebuildsstuff committed:
Add OpenWeather plugin

Arrowana committed:
Use recommended settings in jupiter swap

JoeyKhd committed:
upd: fixed broken API Documentation URL

JoeyKhd committed:
upd: fixed path

Gajesh2007 committed:
Merge branch 'elizaOS:main' into main

bertux committed:
chore: bump version to 0.1.8-alpha.1 for plugin-arthera

Gajesh2007 committed:
Merge old into develop with regenerated pnpm-lock.yaml

Gajesh2007 committed:
Merge develop into main

Gajesh2007 committed:
fix: settings error

fabianhug committed:
Merge pull request #1558 from Gajesh2007/main

feat: added new plugin - zktls - reclaim

augchan42 committed:
Merge branch 'elizaOS:main' into fix/twitter-wildcard-targets-1556

HashWarlock committed:
feat: add remote attestation action

augchan42 committed:
just add needed change to run unit test

shakkernerd committed:
chore: pnpm lock file

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 4 notes

shakkernerd committed:
Merge branch 'develop' into pr-1884

shakkernerd committed:
chore: fix broken pnpm lock file

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: change file extension from rs -> ts

shakkernerd committed:
Merge branch 'develop' into pr-1885

YoungPhlo committed:
docs: Fix timestamp links for Dev School 3

shakkernerd committed:
Merge branch 'develop' into chore/jupiter-swap-recommended-settings

shakkernerd committed:
Merge pull request #1882 from Arrowana/chore/jupiter-swap-recommended-settings

feat: Use recommended settings in jupiter swap

shakkernerd committed:
Merge branch 'develop' into fix-broken-docs-api-url

shakkernerd committed:
Merge pull request #1881 from JoeyKhd/fix-broken-docs-api-url

Fixed broken API Documentation URL

shakkernerd committed:
Merge branch 'develop' into pr-1880

shakkernerd committed:
Merge branch 'develop' into test/client-telegram

shakkernerd committed:
Merge pull request #1879 from ai16z-demirix/test/client-telegram

test: adding test setup for telegram client

YoungPhlo committed:
docs: Fix timestamp links for Dev School 4

shakkernerd committed:
Merge branch 'develop' into pr-1878

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 5 notes

shakkernerd committed:
fix: conflicting Readme.md files

zhourunlai committed:
fix case sensitive path

shakkernerd committed:
chore: explicitly set node version and bump pnpm version

shakkernerd committed:
Merge branch 'develop' into fix_case_sensitive_path

shakkernerd committed:
Merge pull request #1892 from zhourunlai/fix_case_sensitive_path

fix: remove case sensitive path

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
chore: remove pnpm install with no frozen lockfile flag

shakkernerd committed:
chore: pnpm lock file

odilitime committed:
hyperfi endpoint, clean up multer/imports/console logs, notes

shakkernerd committed:
Merge pull request #1896 from odilitime/hfi

feat: hyperfy support

shakkernerd committed:
chore: make viem a global package & remove from plugins package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: add no frozen install flag

odilitime committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
chore: remove flag and disable removing lock file

shakkernerd committed:
chore: changes to packages

shakkernerd committed:
chore: add frozen flag

zhourunlai committed:
fix twitter undefined properties

tomguluson92 committed:
update goplus to the safety module

tomguluson92 committed:
Update README.md

RatakondalaArun committed:
fix: add callback to the evaluators for client-telegram

Commit:
Add Dutch (Nederlands) translation

0xbeekeeper committed:
Update index.ts

lalalune committed:
Merge branch 'develop' into add-dutch-translation

lalalune committed:
Merge pull request #1909 from xzrfax/add-dutch-translation

Add Dutch (Nederlands) translation

lalalune committed:
Merge pull request #1905 from zhourunlai/fix_twitter_undefined_properties

fix: twitter result obj undefined properties

shakkernerd committed:
chore: remove conflict package

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
Revert "chore: enable strict null checks"

This reverts commit fcf913797d7345b3e418c817d88347b5ae87868c.

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1893 from elizaOS/fix/tests

chore: fix integrations and smoke tests

shakkernerd committed:
Merge branch 'develop' into 1906--fix-evaluate-callback-for-client-telegram

bodhi-crypo committed:
Update summarize_conversation.ts

JoeyKhd committed:
upd: resolved a bunch of type related errors and ensure project compiles properly

JoeyKhd committed:
Merge branch 'develop' into fix-types-and-build

Lukapetro committed:
chore: resolve merge conflicts with develop

rawgroundbeef committed:
fix(core): check evaluators for null

shakkernerd committed:
Merge pull request #1908 from RatakondalaArun/1906--fix-evaluate-callback-for-client-telegram

fix: add callback to the evaluators for client-telegram

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1916 from bodhi-crypo/main

chore(client-discord): fix SUMMARIZE prompt

shakkernerd committed:
Merge branch 'develop' into fix/runtime-evaluate-null

shakkernerd committed:
Merge pull request #1918 from bentatum/fix/runtime-evaluate-null

fix(core): check evaluators for null

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1917 from JoeyKhd/fix-types-and-build

fix: resolved a bunch of type related errors and ensure project compiles

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #1907 from tomguluson92/main

feats: diagram update

tcm390 committed:
add followinf option for timeline action

bertux committed:
Merge branch 'develop' into plugin-arthera-alpha

zhourunlai committed:
fix empty input text

tcm390 committed:
Merge branch 'develop' into tcm-max-actions

shakkernerd committed:
Merge pull request #1824 from elizaOS/tcm-max-actions

feat: Optimize Agent Action Processing by Prioritizing Timelines and Limiting Actions Per Cycle

wtfsayo committed:
Merge branch 'develop' into feature/add-binance-plugin

petryshkaCODE committed:
Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge pull request #1812 from Lukapetro/feature/add-binance-plugin

Feature/add binance plugin

wtfsayo committed:
Merge branch 'develop' into Fix-typo

wtfsayo committed:
Merge pull request #1922 from petryshkaCODE/Fix-typo

Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge branch 'develop' into fix_empty_input_text

wtfsayo committed:
Merge branch 'develop' into goplus-security

wtfsayo committed:
Merge pull request #1898 from 0xbeekeeper/goplus-security

feat: add GoPlus Security Plugin to enhance security for agent

wtfsayo committed:
Merge branch 'develop' into plugin-arthera-alpha

JoeyKhd committed:
upd: hide character settings secrets from api route

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

shakkernerd committed:
Merge branch 'develop' into hide-secrets

shakkernerd committed:
Merge pull request #1923 from JoeyKhd/hide-secrets

API route update

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

ai16z-demirix committed:
test: adding tests for discord-client

jonathangus committed:
Merge branch 'develop' into dynamic-template

bertux committed:
Merge branch 'develop' into plugin-arthera-alpha

shakkernerd committed:
Merge pull request #1919 from zhourunlai/fix_empty_input_text

fix: handle empty input text to avoid memory content empty error

shakkernerd committed:
Merge branch 'develop' into tests/client-discord

shakkernerd committed:
Merge pull request #1927 from ai16z-demirix/tests/client-discord

test: adding tests for discord-client

shakkernerd committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
Merge pull request #1894 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #8" notes

proteanx committed:
rm unused variable x_server_url

tcm390 committed:
Limit the number of timelines fetched

tcm390 committed:
make a constant

tcm390 committed:
add comment

sinecose committed:
Merge branch 'develop' into zkevm-no-warning

shakkernerd committed:
Merge pull request #1931 from elizaOS/tcm-limit-fetchingTimelines

fix: Limit the number of timelines fetched

shakkernerd committed:
Merge branch 'develop' into rmxserver

shakkernerd committed:
Merge pull request #1930 from proteanx/rmxserver

fix: rm unused variable X_SERVER_URL

HashWarlock committed:
Merge branch 'develop' into dynamic-template

HashWarlock committed:
Merge pull request #1467 from jonathangus/dynamic-template

feat: Make templates in composeContext dynamic

threewebcode committed:
docs: it is processAction instead of triggerAction

There is only processAction in runtime instance. Update them correctly.

shakkernerd committed:
Merge pull request #1937 from threewebcode/develop

docs: it is processAction instead of triggerAction

Commit:
feat: support for eternalai provider can make request with chain_id extra data in body

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into feat/sol-agent-kit

odilitime committed:
Merge pull request #1373 from renlulu/feat/sol-agent-kit

feat: solana token deploy using solana agent kit

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

bendanzhentan committed:
refactor: remove unused imports

shakkernerd committed:
Merge branch 'develop' into chore/remove-unused-imports

shakkernerd committed:
Merge pull request #1941 from bendanzhentan/chore/remove-unused-imports

chore(plugin-conflux): remove unused imports

shlokkhemani committed:
chore: add embedding tests

ag-wnl committed:
Merge branch 'develop' into zkevm-no-warning

ag-wnl committed:
Merge pull request #1744 from sinecose/zkevm-no-warning

feat(plugin-cronoszkevm): rm not used imports

0xRider committed:
chore: more specific rpc urls

0xRider committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

shakkernerd committed:
Merge pull request #1938 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can make request with chain_id extra data in body

shakkernerd committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

Commit:
feat: support for eternalai provider can write request/response, change name in env

shakkernerd committed:
Merge pull request #1945 from 0xRider/chore/more-specific-rpc-urls

chore: more specific rpc urls

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

shakkernerd committed:
Merge pull request #1948 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can write request/response log info

shakkernerd committed:
Merge branch 'develop' into add-embedding-tests

shakkernerd committed:
Merge pull request #1944 from shlokkhemani/add-embedding-tests

chore: add embedding tests

JoeyKhd committed:
upd: fixed issue with c3po character json

ChristopherTrimboli committed:
Merge remote-tracking branch 'upstream/develop' into chore/dynamic-plugins

wtfsayo committed:
Merge branch 'develop' into plugin-arthera-alpha

JoeyKhd committed:
upd: updated validateCharacterConfig to give better feedback

wtfsayo committed:
Merge pull request #1818 from bertux/plugin-arthera-alpha

feat: new plugin Arthera Chain

lalalune committed:
Merge pull request #1951 from JoeyKhd/fix-c3po

fix: fixed error in C3PO and improved error handling feedback

ChristopherTrimboli committed:
add new plugins

wtfsayo committed:
Merge branch 'develop' into pr/1810

wtfsayo committed:
update .env.example and gitignore

wtfsayo committed:
Merge pull request #1810 from KONFeature/1809-add-pglite-adapter

feat: add pglite db adapter

ChristopherTrimboli committed:
Merge remote-tracking branch 'upstream/develop' into chore/dynamic-plugins

ChristopherTrimboli committed:
Merge remote-tracking branch 'upstream/develop' into chore/dynamic-plugins

ChristopherTrimboli commented:
ready for review, let's stick to just the top of file imports being removed, not overcomplicate this, 1 for 1 these changes are an improvment, nothing different logically, plugins in character configs still do their thing.
